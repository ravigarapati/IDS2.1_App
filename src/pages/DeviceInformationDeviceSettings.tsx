import React, {useEffect} from 'react';
import {Image, View, StyleSheet, TouchableOpacity} from 'react-native';
import {BoschIcon, CustomText} from '../components';
import {Dictionary} from '../utils/dictionary';
import DEVICE_SETTINGS from '../assets/images/DeviceInfoSettings.svg';
import {Icons} from '../utils/icons';

function Info({section, value, title}) {
  return (
    <View
      accessible={true}
      accessibilityLabel={`${title}: ${value}`}
      style={styles.infoContainer}>
      <CustomText
        allowFontScaling={true}
        style={styles.info}
        align={'left'}
        text={`${section}:`}
      />
      <CustomText
        allowFontScaling={true}
        style={styles.info}
        align={'left'}
        text={value}
      />
    </View>
  );
}

export default function DeviceInformationDeviceSettings({navigation}) {
  return (
    <View style={styles.container}>
      <>
        <View style={styles.headerContainer}>
          <View style={styles.headerDivision}>
            <TouchableOpacity
              style={styles.headerBackButton}
              onPress={() => navigation.goBack()}>
              <BoschIcon
                name={Icons.backLeft}
                size={24}
                style={styles.marginHorizontal10}
              />
            </TouchableOpacity>
          </View>

          <View style={styles.headerTitle}>
            <CustomText
              text={'Device Information'}
              size={21}
              allowFontScaling={true}
              font="medium"
              style={{
                marginVertical: 8,
              }}
            />
          </View>
        </View>
        <Image
          style={styles.headerRibbon}
          source={require('../assets/images/header_ribbon.png')}
        />
      </>
      <View style={styles.imageContainer}>
        <DEVICE_SETTINGS fill="#000" />
      </View>

      <CustomText
        allowFontScaling={true}
        style={styles.screenDescription}
        text={Dictionary.bccDashboard.settings.deviceInformationScreen.title}
        accessibilityLabelText={
          Dictionary.bccDashboard.settings.deviceInformationScreen.titleLabel
        }
        font="regular"
      />
      <CustomText
        allowFontScaling={true}
        style={styles.deviceHeader}
        align="left"
        text={Dictionary.bccDashboard.settings.deviceInformationScreen.device}
        accessibilityLabelText={
          Dictionary.bccDashboard.settings.deviceInformationScreen.description
        }
      />
      <View style={styles.marginTop24}>
        <Info
          section={'Model'}
          value={navigation.getParam('deviceInformation').model}
          title={'Device Modal'}
        />
        <Info
          section={'SW Version'}
          value={navigation.getParam('deviceInformation').sw}
          title={'Software version'}
        />
        <Info
          section={'HD Version'}
          value={navigation.getParam('deviceInformation').hd}
          title={'Hardware version'}
        />
        <Info
          section={'MAC ID'}
          value={navigation.getParam('deviceInformation').deviceId}
          title={'MAC ID'}
        />
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {backgroundColor: 'white', flex: 1},
  imageContainer: {
    alignItems: 'center',
    marginTop: 32,
  },
  infoContainer: {
    flexDirection: 'row',
    marginLeft: 17,
    marginBottom: 15,
  },
  info: {
    width: '50%',
  },
  marginTop24: {
    marginTop: 24,
  },
  deviceHeader: {
    backgroundColor: '#EEEEEE',
    paddingVertical: 13,
    paddingLeft: 17,
  },
  screenDescription: {
    marginTop: 23,
    marginBottom: 32,
  },
  marginHorizontal10: {marginHorizontal: 10},
  headerContainer: {
    backgroundColor: '#ffff',
    flexDirection: 'row',
    alignItems: 'center',
  },
  headerDivision: {
    flexDirection: 'column',
    flex: 0.15,
    backgroundColor: '#ffff',
    justifyContent: 'center',
  },
  headerBackButton: {
    justifyContent: 'center',
  },
  headerTitle: {
    flexDirection: 'column',
    flex: 1,
    alignItems: 'flex-start',
    justifyContent: 'center',
    marginLeft: 24,
  },
  headerRibbon: {height: 8, width: '100%'},
});
