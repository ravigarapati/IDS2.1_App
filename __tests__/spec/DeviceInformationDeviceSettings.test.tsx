import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import DeviceInformationDeviceSettings from '../../src/pages/DeviceInformationDeviceSettings';

jest.useFakeTimers();

const mockStore = configureMockStore();
const store = mockStore({
  homeOwner: {
    energyData: {},
    deviceList: [],
    selectedDevice: {},
    contractorMonitoringStatus: false,
    fanMode: '',
    fanOnFor: '',
    fanOffFor: '',
    fanIsScheduled: '',
    fanScheduledStart: '',
    fanScheduledEnd: '',
    selectedSchedule: '0',
    selectedScheduleName: '',
    scheduleInfo: {},
    prevSelectedDevice: {},
    deviceStatus: {},
    deviceInformation: {},
    deviceDetails: {},
    heatPumpInfo: {},
    reloadHeatPumpInfo: false,
    notificationList: [],
    lastKey: {},
    notificationsCount: 0,
    lastUpdated: null,
    isTermsConditionsAccepted: null,
    hoUserAnalytics: null,
    selectedUnitName: {},
    isThermostatSelected: true,
    isUserFirstLogin: true,
    actualWeatherOnFahrenheit: true,
    hapticVibration: true,
    _24HrsFormatSelected: false,
    location: 'Alabaster',
    updatedSelectedDevice: false,
    weatherInfo: {},
    weatherInfoLocation: {},
    locationSuggestions: [],
    locationInformation: {},
    idsSelectedDeviceAccess: '',
    utilyEnergySaving: '',
    idsSelectedDevice: '',
    idsSelectedOdu: '',
    idsSelectedDeviceType: '',
    tempLocation: '',
    newDeviceInfo: {},
    previousBcc: '',
    deviceList2: [
      /*{
        id: 2,
        //macId: 'e8fdf8a4773c',
        macId: '34eae7c351e6',
        deviceName: 'Bedroom Thermostat',
        deviceType: 'BCC100',
        isThermostat: true,
        isOn: true,
        mode: 0,
        setPoint: 74,
        heatingSetpoint: 74,
        coolingSetpoint: 79,
        current: 76,
        isAccesories: true,
        accesories: [
          {
            id: 0,
            name: 'Dehumidifier',
            isEnable: 0,
            isOn: false,
          },
          {
            id: 1,
            name: 'Humidifier',
            isEnable: 1,
            isOn: false,
          },
        ],
        isMonitoring: null,
        isOnSchedule: true,
        stage: 3,
        isConnected: true,
        acceleratedHeating: true,
        schedules: [],
      },*/
    ],
  },
  auth: {},
  contractor: {
    selectedUnit: {
      odu: {
        modelNumber: 'testing',
      },
      systemStatus: 'testing',
      gateway: {gatewayId: 'testing'},
    },
  },
});

describe('BCC Device information screen', () => {
  test('Component to be defined', () => {
    const navState = {params: {deviceInformation: {model: 'BCC101'}}};
    const navigation = {
      getParam: (key, val) => navState?.params[key] ?? val,
    };
    const tree = renderer
      .create(
        <Provider store={store}>
          <DeviceInformationDeviceSettings navigation={navigation} />
        </Provider>,
      )
      .toJSON();

    expect(tree).toBeDefined();
  });
  test('Component to be rendered', () => {
    const navState = {params: {deviceInformation: {model: 'BCC101'}}};
    const navigation = {
      getParam: (key, val) => navState?.params[key] ?? val,
    };
    const tree = renderer
      .create(
        <Provider store={store}>
          <DeviceInformationDeviceSettings navigation={navigation} />
        </Provider>,
      )
      .toJSON();

    expect(tree).toBeDefined();
    expect(tree).toMatchSnapshot();
  });
});
