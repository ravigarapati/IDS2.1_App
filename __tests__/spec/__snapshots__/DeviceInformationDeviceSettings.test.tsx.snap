// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BCC Device information screen Component to be rendered 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffff",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffff",
          "flex": 0.15,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 24,
              },
              Object {
                "marginHorizontal": 10,
              },
              Object {
                "fontFamily": "Bosch-Icon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          î‚ 
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "marginLeft": 24,
        }
      }
    >
      <Text
        accessible={true}
        numberOfLines={0}
        style={
          Array [
            Object {
              "textAlign": "center",
            },
            Object {
              "color": "#000000",
            },
            Object {
              "fontSize": 21,
              "lineHeight": 31.5,
            },
            Object {
              "fontFamily": "BoschSans-Medium",
            },
            Object {
              "marginVertical": 8,
            },
          ]
        }
      >
        Device Information
      </Text>
    </View>
  </View>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/header_ribbon.png",
      }
    }
    style={
      Object {
        "height": 8,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": 32,
      }
    }
  >
    <svg
      data-file-name="SvgDeviceInfoSettings"
      fill="#000"
    />
  </View>
  <Text
    accessibilityLabel="View information about your device"
    accessible={true}
    numberOfLines={0}
    style={
      Array [
        Object {
          "textAlign": "center",
        },
        Object {
          "color": "#000000",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "fontFamily": "BoschSans-Regular",
        },
        Object {
          "marginBottom": 32,
          "marginTop": 23,
        },
      ]
    }
  >
    View information about
your device
  </Text>
  <Text
    accessibilityLabel="Information about the device, model, software version, hardware version and mac id."
    accessible={true}
    numberOfLines={0}
    style={
      Array [
        Object {
          "textAlign": "left",
        },
        Object {
          "color": "#000000",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "fontFamily": "BoschSans-Regular",
        },
        Object {
          "backgroundColor": "#EEEEEE",
          "paddingLeft": 17,
          "paddingVertical": 13,
        },
      ]
    }
  >
    Device
  </Text>
  <View
    style={
      Object {
        "marginTop": 24,
      }
    }
  >
    <View
      accessibilityLabel="Device Modal: BCC101"
      accessible={true}
      style={
        Object {
          "flexDirection": "row",
          "marginBottom": 15,
          "marginLeft": 17,
        }
      }
    >
      <Text
        accessible={true}
        numberOfLines={0}
        style={
          Array [
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "#000000",
            },
            Object {
              "fontSize": 16,
              "lineHeight": 24,
            },
            Object {
              "fontFamily": "BoschSans-Regular",
            },
            Object {
              "width": "50%",
            },
          ]
        }
      >
        Model:
      </Text>
      <Text
        accessible={true}
        numberOfLines={0}
        style={
          Array [
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "#000000",
            },
            Object {
              "fontSize": 16,
              "lineHeight": 24,
            },
            Object {
              "fontFamily": "BoschSans-Regular",
            },
            Object {
              "width": "50%",
            },
          ]
        }
      >
        BCC101
      </Text>
    </View>
    <View
      accessibilityLabel="Software version: undefined"
      accessible={true}
      style={
        Object {
          "flexDirection": "row",
          "marginBottom": 15,
          "marginLeft": 17,
        }
      }
    >
      <Text
        accessible={true}
        numberOfLines={0}
        style={
          Array [
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "#000000",
            },
            Object {
              "fontSize": 16,
              "lineHeight": 24,
            },
            Object {
              "fontFamily": "BoschSans-Regular",
            },
            Object {
              "width": "50%",
            },
          ]
        }
      >
        SW Version:
      </Text>
      <Text
        accessible={true}
        numberOfLines={0}
        style={
          Array [
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "#000000",
            },
            Object {
              "fontSize": 16,
              "lineHeight": 24,
            },
            Object {
              "fontFamily": "BoschSans-Regular",
            },
            Object {
              "width": "50%",
            },
          ]
        }
      >
        Text
      </Text>
    </View>
    <View
      accessibilityLabel="Hardware version: undefined"
      accessible={true}
      style={
        Object {
          "flexDirection": "row",
          "marginBottom": 15,
          "marginLeft": 17,
        }
      }
    >
      <Text
        accessible={true}
        numberOfLines={0}
        style={
          Array [
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "#000000",
            },
            Object {
              "fontSize": 16,
              "lineHeight": 24,
            },
            Object {
              "fontFamily": "BoschSans-Regular",
            },
            Object {
              "width": "50%",
            },
          ]
        }
      >
        HD Version:
      </Text>
      <Text
        accessible={true}
        numberOfLines={0}
        style={
          Array [
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "#000000",
            },
            Object {
              "fontSize": 16,
              "lineHeight": 24,
            },
            Object {
              "fontFamily": "BoschSans-Regular",
            },
            Object {
              "width": "50%",
            },
          ]
        }
      >
        Text
      </Text>
    </View>
    <View
      accessibilityLabel="MAC ID: undefined"
      accessible={true}
      style={
        Object {
          "flexDirection": "row",
          "marginBottom": 15,
          "marginLeft": 17,
        }
      }
    >
      <Text
        accessible={true}
        numberOfLines={0}
        style={
          Array [
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "#000000",
            },
            Object {
              "fontSize": 16,
              "lineHeight": 24,
            },
            Object {
              "fontFamily": "BoschSans-Regular",
            },
            Object {
              "width": "50%",
            },
          ]
        }
      >
        MAC ID:
      </Text>
      <Text
        accessible={true}
        numberOfLines={0}
        style={
          Array [
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "#000000",
            },
            Object {
              "fontSize": 16,
              "lineHeight": 24,
            },
            Object {
              "fontFamily": "BoschSans-Regular",
            },
            Object {
              "width": "50%",
            },
          ]
        }
      >
        Text
      </Text>
    </View>
  </View>
</View>
`;
